injector_helper.c

```diff
@@ -88,11 +88,10 @@ static void handle_injector_response(struct evhttp_request *res, void *ctx_void)
 static void forward_request_to_injector(request_ctx *ctx, injector *i);
 
 // Returns true on error
-static bool fd_local_info(int fd, const char **addr, uint16_t *port)
+static bool fd_local_info(int fd, char *addr, size_t addrlen, uint16_t *port)
```

Comment:
What does this function do? Where does it change addr? What does addr represent?